app/controllers/homebuyings_controller.rb

```diff
@@ -3,79 +3,110 @@ class HomebuyingsController < ApplicationController
   respond_to :html, :json
 
   def index
-    @homebuyings = Homebuying.all
+    if current_client || current_user
+      @homebuyings = Homebuying.all
+    else
+      redirect_to "/"
+    end
   end
 
   def new
-    @homebuying = Homebuying.new
+    if current_client
+      @homebuying = Homebuying.new
+    else
+      redirect_to "/"
+    end
   end
 
-  def create
-    @id = current_client.id if current_client
-    @id = Client.find(params[:id]).id if current_user
 
-    @homebuying = Homebuying.new({
-      client_id: @id,
-      lender: params[:lender],
-      hear_of_workshop: params[:hear_of_workshop],
-      contact_for_appointment: params[:contact_for_appointment],
-      real_estate_contract: params[:real_estate_contract],
-      realtor_name: params[:realtor_name],
-      realtor_phone: params[:realtor_phone],
-      property_address: params[:property_address],
-      property_state: params[:property_state],
-      property_city: params[:property_city],
-      loan_officer_name: params[:loan_officer_name],
-      loan_officer_email: params[:loan_officer_email],
-      loan_officer_phone: params[:loan_officer_phone],
-      payment_assistance_program: params[:payment_assistance_program],
-      approx_closing_date: params[:approx_closing_date],
-    })
+  def create
+    if current_client || current_user
```

Comment:
This if clause isn't needed because the @id only gets assigned if thats the case. The only other thing it could be doing is verifying they are logged in to allow the rest of the method to work, which should be handled by a before_action.